proposals/0197-chilli-peppers.md

```diff
@@ -0,0 +1,147 @@
+These clocks are also implemented as 64-bit unsigned integers. Whenever a hot account is read from or written to within a block, its Account State Unit Clock is updated to match the current Block State Unit Clock.
+All accounts will have a Chili Pepper Clock Timestamp, which will only exist for accounts which are hot, and is discarded for accounts which are cold. All implementations should keep track of the current hot accounts.
+ 
+#### Hot and Cold Account Designation
+ 
+An account is designated as cold when its Account Chili Pepper Clock Timestamp falls behind the current Block Chili Pepper Clock by more than the Hot Cache Size parameter.
+An account which has never existed is considered cold. An account that is deleted is still considered hot until its state unit clock lapses into cold. Creating an account against a
+deleted account which is still hot, will create the hot account again.
+ 
+#### Storage and Management
+ 
+To manage the Account State Unit Clocks efficiently, Solana employs a table, associating each hot account with its respective Chili Pepper Clock Timestamp. This table enables the dynamic tracking and updating of account states, facilitating the transition of accounts between hot and cold statuses based on their activity.
+ 
+### Error Cases for State Units Implementation
+ 
+Here are common error scenarios related to State Units and their respective handling mechanisms:
+ 
+#### Exceeding Block Chili Pepper Limit
+ 
+- **Error Description**: This error occurs when a transaction's requested Chili Peppers exceed the remaining capacity of the Block Chili Pepper Clock for the current block.
+- **Handling**: The block is marked as invalid and cannot be processed.
+ 
+#### Invalid Chili Pepper Request
+ 
+- **Error Description**: A transaction specifies an invalid number of Chili Peppers, either by requesting more than a predefined maximum limit per transaction or by formatting the request improperly.
+- **Handling**: The transaction is invalidated, and an "Invalid Chili Pepper Request" error message is issued. Developers must ensure that Chili Pepper requests conform to protocol specifications, including proper formatting and adherence to maximum limits.
+ 
+#### Account State Unit Clock Synchronization Failure
+ 
+#### Accessing Cold Account
+ 
+- **Error Description**: A transaction attempts to interact with an account that has been designated as "cold" due to its Chili Pepper Clock Timestamp falling below the "Hot Cache Size" threshold relative to the current Block Chili Pepper Clock, without requesting sufficient Chili Peppers.
+- **Handling**: The transaction is rejected with a "Cold Account Access Attempted" error.
+ 
+ 
+## Alternatives Considered
+ 
+What alternative designs were considered and what pros/cons does this feature
+have relative to them?
+ 
+## Impact
+ 
+How will the implemented proposal impacts dapp developers, validators, and core contributors?
```

Comment:
We will need to expand this section. 
How will this impact dapp developers?

Reply:
calculating chilli peppers could be mostly hidden by the client sdk so I would guess once people update to them, there should be almost zero effect

Reply:
> every transaction on the Solana network will be required to use the ComputeBudget program to request the maximum number of Chili Peppers they require.

Wouldn't the dapp specify the Chili pepper in the transactions? In order to do so, will they need to know whether the account is hot or cold?